From 661482c504b9f1d79d5811d7b4ba99f4c7ce5108 Mon Sep 17 00:00:00 2001
From: Martin Jansa <martin2.jansa@lgepartner.com>
Date: Sat, 11 Mar 2023 22:26:22 +0000
Subject: [PATCH] CMakeLists.txt: update from libprocps to libproc2

:Release Notes:

:Detailed Notes:
Fixes build with procps-4:
http://gecko.lge.com:8000/Errors/Details/551376

-- Checking for module 'libprocps'
--   No package 'libprocps' found

there might be many more changes needed to adapt to procps 4 API, see:
https://gitlab.com/procps-ng/procps/-/issues/239

procps upgrade was now merged in oe-core together with igt-gpu-tools adaptation in:
https://git.openembedded.org/openembedded-core/commit/?id=631eba02d64f1a42514e0ae4361bbecc5cce5fa7

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRP-10874] Prepare for procps 4 API

Change-Id: Ifd45a298f64afd4550fb3384926a582dcb1c6a50
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/com.webos.service.unifiedsearch/+/347406 CMakeLists.txt: update from libprocps to libproc2]

 src/CMakeLists.txt        |  9 +++++++--
 src/util/LinuxProcess.cpp | 11 ++++++++++-
 2 files changed, 17 insertions(+), 3 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 23bb851..a21772d 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -27,7 +27,12 @@ pkg_check_modules(PBNJSON REQUIRED pbnjson_cpp)
 include_directories(${PBNJSON_INCLUDE_DIRS})
 webos_add_compiler_flags(ALL ${PBNJSON_CFLAGS_OTHER})
 
-pkg_check_modules(PROCPS REQUIRED libprocps)
+pkg_check_modules(PROCPS libproc2)
+if(PROCPS_FOUND)
+    add_definitions(-DHAS_LIBPROC2=1)
+else()
+    pkg_check_modules(PROCPS libprocps)
+endif()
 include_directories(PROCPS_INCLUDE_DIRS)
 webos_add_compiler_flags(ALL ${PROCPS_CFLAGS})
 
@@ -62,4 +67,4 @@ install(TARGETS ${BIN_NAME} DESTINATION ${WEBOS_INSTALL_SBINDIR})
 webos_build_system_bus_files()
 
 webos_build_daemon(NAME ${BIN_NAME})
-webos_build_configured_file(files/conf/unifiedsearch.json WEBOS_SYSCONFDIR "")
\ No newline at end of file
+webos_build_configured_file(files/conf/unifiedsearch.json WEBOS_SYSCONFDIR "")
diff --git a/src/util/LinuxProcess.cpp b/src/util/LinuxProcess.cpp
index 1737064..e41ec9b 100644
--- a/src/util/LinuxProcess.cpp
+++ b/src/util/LinuxProcess.cpp
@@ -16,7 +16,6 @@
 
 #include <signal.h>
 #include <glib.h>
-#include <proc/readproc.h>
 #include <stdlib.h>
 #include <unistd.h>
 #include <pwd.h>
@@ -24,6 +23,12 @@
 
 #include "util/LinuxProcess.h"
 
+#if HAS_LIBPROC2
+#include <libproc2/pids.h>
+#else
+#include <proc/readproc.h>
+#endif
+
 const string LinuxProcess::CLASS_NAME = "LinuxProcess";
 
 bool LinuxProcess::sendSigTerm(const string& pid)
@@ -178,6 +183,9 @@ PidVector LinuxProcess::findChildPids(const string& pid)
     PidVector pids;
     pids.push_back((pid_t) atol(pid.c_str()));
 
+#if HAS_LIBPROC2
+#warning TODO migrate to proc2
+#else
     proc_t **proctab = readproctab(PROC_FILLSTAT);
     if (!proctab) {
         Logger::error(CLASS_NAME, __FUNCTION__, "readproctab_error", "failed to read proctab");
@@ -201,6 +209,7 @@ PidVector LinuxProcess::findChildPids(const string& pid)
     }
 
     free(proctab);
+#endif
     return pids;
 }
 
